sale_order_lot_mrp/models/mrp.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+#  © 2014 ToDay Akretion
+#  @author David BEAL <david.beal@akretion.com>
+#  @author Sébastien Beau <sebastien.beau@akretion.com>
+#  @author Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import models, api, fields
+
+
+class MrpProduction(models.Model):
+    """ Purpose to generate manufacturing base on custom product raw material
+    """
+    _inherit = 'mrp.production'
+
+    lot_id = fields.Many2one(
+        'stock.production.lot',
+        'Lot',
+        index=True)
+
+    @api.multi
+    def button_plan(self):
+        """
+        We passe the production.id in the context so you can easily
+        retrieve the production information during the process that generate
+        the line of needed for the production order. For exemple, in
+        the method explode() of 'mrp.bom', you can read the information
+        of the production order to customise the needed.
+        """
+        for production in self:
+            production = production.with_context(production=production)
+            super(MrpProduction, production).button_plan()
+
+    def _generate_finished_moves(self):
+        move = super(MrpProduction, self).\
+            _generate_finished_moves()
+        move.restrict_lot_id = self.lot_id.id
+        return move
+
+
+class ProcurementOrder(models.Model):
```

Comment:
It would be nice to have separate files for the different models. Not necessarily one model per file, but in any case, ProcurementOrder should not be in mrp.py file.